src/PeriodEnd/PaymentsDue/SFA.DAS.ProviderPayments.Calc.PaymentsDue/PaymentsDueProcessor.cs

```diff
@@ -333,15 +334,15 @@ private void ApportionPaymentDuesOverPreviousPeriods(Provider provider, List<Req
                     AddRefundPaymentDue(provider, paymentsDue, new DateTime(transaction.CollectionPeriodYear, transaction.CollectionPeriodMonth, 1), transaction.AmountDue, transaction.CommitmentId, earning, paymentHistory, amountDue);
                     refundAdded = true;
                 }
-                
+
             }
             if (!refundAdded)
             {
 
                 var refundablePeriods = paymentHistory.Where(x => x.DeliveryMonth == earning.CalendarMonth
                                                                 && x.DeliveryYear == earning.CalendarYear
                                                                 && x.AmountDue > 0
-                                                                && new DateTime(x.CollectionPeriodYear,x.CollectionPeriodMonth,1) !=
+                                                                && new DateTime(x.CollectionPeriodYear, x.CollectionPeriodMonth, 1) !=
```

Comment:
Pull out the current year DateTime creation into a variable as it is otherwise created for each and every member in the where clause. Also, as we are also searching for new DateTime(x.CollectionPeriodYear, x.CollectionPeriodMonth, 1) AND ordering on it, consider altering the where clause to create a variable to represent this to reduce complexity.